### DISCUSSION TITLE
Where to find JSON Schema generated in the docs

### DISCUSSION THREAD

**Discussion by gmsantos** [concourse/concourse#6780] [Help & Support] [answered]
Hey, I looking for the JSON Schema from Concourse pipelines and tasks to include in my IDE.

I found [this one](https://github.com/cappyzawa/concourse-pipeline-jsonschema) but it is not on par with the schemas on the docs (see https://github.com/cappyzawa/concourse-pipeline-jsonschema/issues/3).

I'm wondering if it is possible to get the schema for docs and parse it as a JSON Schema like suggested here https://github.com/concourse/docs/issues/381

The next step would be to publish those schemas to https://www.schemastore.org/json/, giving automatically code validation for IDE using n YAML language server plugin, like [this one](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) for VSCode.

**Comment by taylorsilva** [comment, marked as answer]
We haven't produced a json schema doc. Is it possible to generate one from the golang structs in the code? The code is the source of truth. 

From a quick search
- https://github.com/alecthomas/jsonschema
- https://github.com/urakozz/go-json-schema-generator

Seems possible and somewhat easy. `steps` may be weird though... https://github.com/concourse/concourse/blob/master/atc/steps.go

**Reply by taylorsilva** [reply]
The way we generate our docs is different from most projects. The schema has essentially been re-typed in the docs: https://github.com/concourse/docs/blob/master/lit/docs/jobs/steps.lit

**Reply by gmsantos** [reply]
I would give it a try, thanks @taylorsilva
